### PR TITLE
Rule cache fixes

### PR DESCRIPTION
**Pull request by ralphtheninja**
This PR contains:

* a bug fix for the rule cache
* ~~extra `.close()` methods for propagating down to the rule cache when the server "stops"~~
* keep a reference to the timer in `RuleCache` so it can be accessed

### CODE REVIEW

**Comment by ralphtheninja** [comment]
Unit tests pass but not sure why travis complains.

File: src/permission/rule-cache.js [hunk 1]
```diff
@@ -10,7 +10,7 @@
 const RuleCache = function (options) {
   this._options = options
   this._data = {}
-  setInterval(this._purge.bind(this), options.cacheEvacuationInterval)
+  this._interval = setInterval(this._purge.bind(this), options.cacheEvacuationInterval)
```
Comment (ralphtheninja) [inline src/permission/rule-cache.js]: This should not affect anything _but_ the server can at least be closed properly by the caller by doing `clearInterval()` from the outside.

File: src/permission/config-permission-handler.js [hunk 2]
```diff
@@ -33,9 +33,9 @@ const UNDEFINED = 'undefined'
  *                           to load it from the path provided in options.path.
  */
 const ConfigPermissionHandler = function (options, config) {
-  this._ruleCache = new RuleCache(options)
```
Comment (ralphtheninja) [inline src/permission/config-permission-handler.js]: Passing `options` here is incorrect. `RuleCache` actually wants `options.permission.options`, or `this._permissionOptions`.

**Comment by yasserf** [comment]
Very good catch! That's a little embarrassing. We'll try get the test to pass with clearing the interval to allow the process to die gracefuly

**Comment by ralphtheninja** [comment]
> That's a little embarrassing.

Not easy to catch this. I suspect it's related to changes in configuration, like re-grouping parameters or something like that.

**Comment by AlexBHarley** [comment]
Want to get this into the next release so merging now, we'll get the graceful shutdown in as part of a bigger story later.

**Comment by ralphtheninja** [comment]
FYI. We run an older version of deepstream (before this fix) on one server and the cpu (`docker stats`) is on around 4% when nothing happens. After this fix it goes down to under 0.1%

**Pull request merged**